addons/website/static/src/snippets/s_accordion/000.scss

```diff
@@ -20,6 +20,98 @@
         }
     }
 
+    .o_icons_side_to_bottom {
+        --accordion-btn-icon-transform: rotate(0deg);
+
+        .accordion-button.collapsed {
+            &.o_icons_position_reversed:after {
+                transform: rotate(-90deg);
+            }
+
+            &:after {
+                transform: rotate(90deg);
+            }
+        }
+    }
+
+    .o_custom_icons_wrap {
+        display: none;
+        width: $accordion-icon-width;
+        height: $accordion-icon-width;
+
+        > span {
+            position: absolute;
+            inset: 0;
+            display: none;
+        }
+    }
```

Comment:
Pretty option but it should definitely not be implemented this way: you have extra invisible DOM to load in the default case with extra CSS to load just to hide it. Plus you don't create a new version of the accordion (nice) but I doubt old accordions would not be broken since you seem to rely on the fact this new structure is there.

IMO this should be adapted so that the JS option creates that `o_custom_icons_wrap` element when selecting custom icons and removes it when going back to default. That ensures old accordion still work and receive the new option without any need of an upgrade but also optimizes the DOM of the default snippet.

With the amazing training you just got, you should be able to do it? :laughing: But if you want, we can delegate that to someone here.

Reply:
> Small note: I've created a private function (_updateCustomIcons) used in the initial customIcon function to make the code easier to read and avoid having to define a .forEach() for each condition.

This definitely makes it **harder** to read to me :grimacing: I have to jump to that private function 4 times, the function does things at the start that is only required for the 3rd call and I had to understand your "action" thing. Plus this is very inefficient to have to recheck the condition each turn of the forEach. I assure you this does not make things easier to read :wink: 

Duplicating the forEach is better but in the end, I don't think you even need that. For the `addCustomIcons` and `removeCustomIcons` "actions", you can just put it at the end of your first forEach, no? Something like `customIconWrapEl.innerHTML = customIcons`, and outside of the forEach, you first set `customIcons` to the right value (empty string or the custom icons) depending on the situation.

>  Also I just have now a red Security check on runbot 🤔 .

Yes, that's because `innerHTML =` might be unsafe. When you can, it's better to append elements, that you can render with `renderToElement` or `renderToFragment`. In this case... might not be the best course of action as you would need to re-render X times or duplicate the rendered element/fragement. Don't worry about this, I'll decide when I'll merge if I change it or if just ask for the security bypass :)

Reply:
Oh ok I see 🤔 
Alright, I updated the script based on your suggestion. What do you think ? 🙂